app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -202,7 +202,9 @@ class TransactionStateManager extends EventEmitter {
     const previousState = replayHistory(txMeta.history)
     // generate history entry and add to history
     const entry = generateHistoryEntry(previousState, currentState, note)
-    txMeta.history.push(entry)
+    if (entry[0]) {
```

Comment:
Why `entry[0]` and not `entry.length`? Could it return e.g. `[ null ]` or `[ undefined ]`?

Reply:
Fair enough, I was mimicking what `generateHistoryEntry` was doing internally. That said, both could use `entry.length`.

Reply:
Updated